src/main/kotlin/calculator/domain/Operator.kt

```diff
@@ -0,0 +1,15 @@
+package calculator.calculator.domain
+
+enum class Operator(private val symbol: String, private val operate: (Double, Double) -> Double) {
+    ADD("+", { prev, next -> prev + next }),
+    SUB("-", { prev, next -> prev - next }),
+    MUL("*", { prev, next -> prev * next }),
+    DIV("/", { prev, next -> prev / next });
+
+    companion object {
+        fun from(symbol: String) = values().find { op -> op.symbol == symbol }
+            ?: throw IllegalArgumentException("존재하지 않는 기호입니다.")
```

Comment:
혹시 require을 사용한다면 어떻게 사용하는게 적절했을까요?

```kotlin
values().find { op -> op.symbol == symbol }.also { requireNotNull(it) {"존재하지 않는 기호입니다."} }
```
이렇게 시도했었는데 T?타입을 반환하더군요
이렇게 T타입이 필요한데 T?타입을 반환하는 경우가 흔할 것 같은데 일반적으로 어떻게 처리하나요?

-----
아래와 같이 처리할 수 있습니다.
```suggestion
        fun from(symbol: String): Operator =
            requireNotNull(values().find { it.symbol == symbol }) { "존재하지 않는 기호입니다." }
```

`T?` 타입이 반환된 이유는 `find()`를 호출하면 `T?`가 반환되고 also() 함수는 호출자를 그대로 반환하기 때문에 `T?.also{ ... }`을 호출했을 때 `T?`이 반환된 것입니다.

-----
이와는 별개로 NotNull 변수가 필요한데 Nullable 변수가 나오는 경우에는 null 자체가 비즈니스에 연관되어 있는 경우와 연관되지 않은 경우로 분리해서 처리합니다.
연관된 경우에는 View까지 null을 전해서 유저에게 어떤 오류가 발생했는지를 알려줍니다.
연관되지 않은 경우에는 바로 무시처리하거나 예외를 던져서 NotNull 변수로 만들기도 합니다.

Reply:
상세한 답변 정말 감사합니다! 많은 도움이 되었습니다!